circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/event/LoggingListener.java

```diff
@@ -93,22 +127,24 @@ public void circusTrainStartUp(String[] args, EventSourceCatalog sourceCatalog,
 
   @Override
   public void resolvedMetaStoreSourceTable(EventTable table) {
-    replicationState.partitionKeys = table.getPartitionKeys();
+    replicationState.setPartitionKeys(table.getPartitionKeys());
   }
 
   @Override
   public void partitionsToCreate(EventPartitions partitions) {
-    replicationState.partitionsAltered += partitions.getEventPartitions().size();
+    replicationState
```

Comment:
If we do decide to stick with this approach then having an `incrementPartitionsAltered` method which takes the new size and then adds it internally to the new size would probably be easier to use.

Reply:
I think Patrick makes a good point about it being more readable before. But it sounds like a good idea, so should I still add in with an `incrementPartitionsAltered` method?

Reply:
Sure.